src/main/java/org/skriptlang/skript/common/function/Parameter.java

```diff
@@ -70,6 +86,100 @@ static Modifier of() {
 		 */
 		Modifier KEYED = of();
 
+		/**
+		 * A modifier to use for checking if a parameter is ranged.
+		 * Do NOT use for declaring a parameter to be ranged, use {@link Modifier#ranged(Comparable, Comparable)}
+		 * Accessing the min and the max values can be done via {@link Parameter#getModifier(Class)}.
+		 */
+		Modifier RANGED = new RangedModifier<>(0,0); // 0 and 0 are just dummy values.
+
+		/**
+		 * Creates a range modifier with inclusive min and max bounds.
+		 */
+		static <T extends Comparable<T>> RangedModifier<T> ranged(T min, T max) {
+			return new RangedModifier<>(min, max);
+		}
+
+		/**
+		 * Modifier specifying valid range bounds for numeric parameters.
+		 * Note that ALL instances will have the same hashCode and will be equal to {@link Modifier#RANGED}.
+		 * Avoid comparing these objects or putting multiple into a HashSet or HashMap!
+		 */
+		class RangedModifier<T extends Comparable<T>> implements Modifier {
+			private final T min;
+			private final T max;
+
+			/**
+			 * Inclusive range between min and max
+			 * @param min min value
+			 * @param max max value
+			 */
+			private RangedModifier(T min, T max) {
+				Preconditions.checkState(min.compareTo(max) < 1, "Min value cannot be greater than max value!");
+				this.min = min;
+				this.max = max;
+			}
+
+			/**
+			 * @return Min value of the range (inclusive)
+			 */
+			public T getMin() { return min; }
+
+			/**
+			 * @return Max value of the range (inclusive)
+			 */
+			public T getMax() { return max; }
+
+			/**
+			 * @param input The value to test.
+			 * @return Whether input is between min and max.
+			 */
+			public boolean inRange(Object input) {
+				return inRange(new Object[]{ input });
```

Comment:
I think it would make more sense for this method to have the single object check. The other method can make use of it.